tests/dynamic_data/users_roles/delete_user_test.py

```diff
@@ -194,6 +198,30 @@ def test_delete_shared_publications_with_readers(setup_user_or_everyone, publica
     process_client.delete_workspace_publications(publication_type, public_workspace, actor_name=username)
 
 
+@pytest.mark.parametrize('publication_type', process_client.PUBLICATION_TYPES)
+@pytest.mark.parametrize('workspace', ["public_workspace", "test_delete_owner_reader_role"])
+@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
+def test_delete_shared_publication_with_role(publication_type, workspace):
```

Comment:
I suggest to join it with `test_delete_own_public_publications` and `test_delete_user` with parametrization of publication workspace and access_rights.